ipinfo/config.go

```diff
@@ -32,74 +32,127 @@ func getConfigDir() (string, error) {
 	return confDir, nil
 }
 
-// init function
+// returns the path to the config file.
+func ConfigPath() (string, error) {
+	confDir, err := getConfigDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(confDir, "config.json"), nil
+}
+
+// returns the path to the token file.
+func TokenPath() (string, error) {
+	confDir, err := getConfigDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(confDir, "token"), nil
+}
+
 func InitConfig() error {
-	path, err := ConfigPath()
+	configpath, err := ConfigPath()
+	if err != nil {
+		return err
+	}
+
+	tokenpath, err := TokenPath()
 	if err != nil {
 		return err
 	}
 
-	if !lib.FileExists(path) {
-		err := SetConfig(NewConfig())
+	// create default config if none yet.
+	if !lib.FileExists(configpath) {
+		gConfig = NewConfig()
+	} else {
+		config, err := ReadConfig()
 		if err != nil {
 			return err
 		}
+		gConfig = config
 	}
-	return nil
-}
 
-// returns the path to config file.
-func ConfigPath() (string, error) {
-	confDir, err := getConfigDir()
-	if err != nil {
-		return "", err
+	// if token exists, migrate it to config file.
```

Comment:
One thing you may notice is that this case is only possible if the user has no `config.json` yet - if they already have it, that means we also already migrated it before. So this logic can actually be pulled up into the `if/else` clause above, in particular in the `if` branch, because that's the first time we generate `config.json` for a user, and the **only** time they may still have the token file.